Bottom: ef1d63f62db8ebdb1838ab82195f82e6a5753788
Top:    0808fe70a4982fe786204c53f00c6418823b9b2d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-08 23:39:43 +0000

Revert "Stop "find connected" following rat-lines. Fixes confusing output."

This reverts commit 764c3560a722c768a7048f5c70811ec363862882.


---

diff --git a/src/action.c b/src/action.c
index a70e4dd..5fa49dc 100644
--- a/src/action.c
+++ b/src/action.c
@@ -872,7 +872,7 @@ NotifyLine (void)
 			       PIN_TYPE | PAD_TYPE | VIA_TYPE, &ptr1, &ptr2,
 			       &ptr3);
 	  LookupConnection (Crosshair.X, Crosshair.Y, true, 1,
-			    FOUNDFLAG, true);
+			    FOUNDFLAG);
 	}
       if (type == PIN_TYPE || type == VIA_TYPE)
 	{
@@ -2288,7 +2288,7 @@ ActionConnection (int argc, char **argv, Coord x, Coord y)
 	case F_Find:
 	  {
 	    gui->get_coords (_("Click on a connection"), &x, &y);
-	    LookupConnection (x, y, true, 1, FOUNDFLAG, false);
+	    LookupConnection (x, y, true, 1, FOUNDFLAG);
 	    break;
 	  }
 
@@ -2754,7 +2754,7 @@ ActionDisplay (int argc, char **argv, Coord childX, Coord childY)
 	      if (Crosshair.AttachedLine.State != STATE_FIRST)
 		LookupConnection (Crosshair.AttachedLine.Point1.X,
 				  Crosshair.AttachedLine.Point1.Y, true, 1,
-				  FOUNDFLAG, true);
+				  FOUNDFLAG);
 	    }
 	  notify_crosshair_change (true);
 	  break;
diff --git a/src/find.c b/src/find.c
index 5318a52..a6e003d 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3221,8 +3221,7 @@ ListStart (int type, void *ptr1, void *ptr2, void *ptr3)
  * also the action is marked as undoable if AndDraw is true
  */
 void
-LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
-                  bool AndRats)
+LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag)
 {
   void *ptr1, *ptr2, *ptr3;
   char *name;
@@ -3236,9 +3235,10 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
     = SearchObjectByLocation (LOOKUP_FIRST, &ptr1, &ptr2, &ptr3, X, Y, Range);
   if (type == NO_TYPE)
     {
-      type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
-        &ptr1, &ptr2, &ptr3, X, Y, Range);
+      type
+        =
+        SearchObjectByLocation
+        (LOOKUP_MORE, &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
       if (type & SILK_TYPE)
@@ -3251,9 +3251,11 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
             return;
         }
     }
-
-  name = ConnectionName (type, ptr1, ptr2);
-  hid_actionl ("NetlistShow", name, NULL);
+  else
+    {
+      name = ConnectionName (type, ptr1, ptr2);
+      hid_actionl ("NetlistShow", name, NULL);
+    }
 
   TheFlag = which_flag;
   User = AndDraw;
@@ -3263,7 +3265,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
    * This is step (1) from the description
    */
   ListStart (type, ptr1, ptr2, ptr3);
-  DoIt (AndRats, AndDraw);
+  DoIt (true, AndDraw);
   if (User)
     IncrementUndoSerialNumber ();
   User = false;
diff --git a/src/find.h b/src/find.h
index ab4021a..6ccc81f 100644
--- a/src/find.h
+++ b/src/find.h
@@ -51,7 +51,7 @@ bool ArcPadIntersect (ArcType *, PadType *);
 bool IsPolygonInPolygon (PolygonType *, PolygonType *);
 void LookupElementConnections (ElementType *, FILE *);
 void LookupConnectionsToAllElements (FILE *);
-void LookupConnection (Coord, Coord, bool, Coord, int, bool AndRats);
+void LookupConnection (Coord, Coord, bool, Coord, int);
 void LookupUnusedPins (FILE *);
 bool ResetFoundLinesAndPolygons (bool);
 bool ResetFoundPinsViasAndPads (bool);
diff --git a/src/netlist.c b/src/netlist.c
index 4a17eec..d47f7d2 100644
--- a/src/netlist.c
+++ b/src/netlist.c
@@ -164,7 +164,7 @@ netlist_find (LibraryMenuType * net, LibraryEntryType * pin)
   int x, y;
   if (pin_name_to_xy (net->Entry, &x, &y))
     return;
-  LookupConnection (x, y, 1, 1, FOUNDFLAG, true);
+  LookupConnection (x, y, 1, 1, FOUNDFLAG);
 }
 
 static void
@@ -173,7 +173,7 @@ netlist_select (LibraryMenuType * net, LibraryEntryType * pin)
   int x, y;
   if (pin_name_to_xy (net->Entry, &x, &y))
     return;
-  LookupConnection (x, y, 1, 1, SELECTEDFLAG, true);
+  LookupConnection (x, y, 1, 1, SELECTEDFLAG);
 }
 
 static void
diff --git a/src/report.c b/src/report.c
index fa33060..64db36a 100644
--- a/src/report.c
+++ b/src/report.c
@@ -560,7 +560,7 @@ XYtoNetLength (Coord x, Coord y, int *found)
   /* NB: The third argument here, 'false' ensures LookupConnection
    *     does not add its changes to the undo system.
    */
-  LookupConnection (x, y, false, PCB->Grid, FOUNDFLAG, true);
+  LookupConnection (x, y, false, PCB->Grid, FOUNDFLAG);
 
   ALLLINE_LOOP (PCB->Data);
   {
